### PR TITLE
4.x dev

### PR DESCRIPTION
**Pull request by ghost**
Copying the changes from #510 to composer package branch.

### CODE REVIEW

**Comment by zbcchen** [comment]
@bzvestey Hi, I am working on this PR. If you have any suggest, let me know, please.

File: src/Saml2/Auth.php [hunk 1]
```diff
@@ -264,11 +265,11 @@ public function processResponse($requestId = null)
     /**
      * Process the SAML Logout Response / Logout Request sent by the IdP.
      *
-     * @param bool        $keepLocalSession             When false will destroy the local session, otherwise will keep it
```
Comment (bzvestey) [inline src/Saml2/Auth.php:267]: Why all of the whitespace changes throughout the files?

File: tests/bootstrap.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 include ('compatibility.php');
 
-ob_start();
```
Comment (bzvestey) [inline tests/bootstrap.php:8]: Why was this commented out?

File: tests/src/OneLogin/Saml2/ResponseTest.php [hunk 3]
```diff
@@ -1534,7 +1533,7 @@ public function testIsInValidCert()
         $response = new Response($settings, $xml);
 
         $this->assertFalse($response->isValid());
-        $this->assertEquals('openssl_x509_read(): supplied parameter cannot be coerced into an X509 certificate!', $response->getError());
```
Comment (bzvestey) [inline tests/src/OneLogin/Saml2/ResponseTest.php:1537]: Any particular reason this was updated?

**Review by bzvestey — CHANGES_REQUESTED** [review]
A couple things here:
- there seems to be a lot of random whitespace changes.
- You have a few other changes that don't seem to be connected with your changes.
- The changes somehow broke a bunch of the tests, so you will need to understand why that is.

File: src/Saml2/Auth.php (hunk 1 above)
Comment (zbcchen) [reply src/Saml2/Auth.php:267]: I will fix them and push again.

**Comment by zbcchen** [comment]
I have fixed them and pushed the changes again. you could take a look if you have time. @bzvestey

File: src/Saml2/version.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
     "php-saml": {
-        "version": "4.0.0",
-        "released": "02/03/2021"
+        "version": "4.0.1",
```
Comment (bzvestey) [inline src/Saml2/version.json:3]: Odd that this is showing up in the diff... since this is what is already is...

**Review by bzvestey — APPROVED** [review]

**Comment by zbcchen** [comment]
> I will aim to release this on Friday.

That's great!

**Pull request merged**